packages/eslint-plugin/tests/rules/prefer-promise-reject-errors.test.ts

```diff
@@ -266,6 +266,7 @@ ruleTester.run('prefer-promise-reject-errors', rule, {
       declare const foo: PromiseConstructor;
       foo.reject(new Error());
     `,
+    'console[Symbol.iterator]();',
```

Comment:
[Testing] This is just a bit different from the `public [Symbol.iterator]() { ... }` from #9999. Could you include that too, please?

Reply:
> You work fast! 😄

Under promise, over deliver, etc etc 😉 

Additional test added!